manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/ProcessorWorker.java

```diff
@@ -0,0 +1,98 @@
+package com.redhat.service.bridge.manager.workers.resources;
+
+import java.util.Objects;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.dao.ConnectorsDAO;
+import com.redhat.service.bridge.manager.dao.ProcessorDAO;
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+import com.redhat.service.bridge.manager.models.Processor;
+import com.redhat.service.bridge.manager.models.Work;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+import io.quarkus.vertx.ConsumeEvent;
+
+@ApplicationScoped
+public class ProcessorWorker extends AbstractWorker<Processor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProcessorWorker.class);
+
+    @Inject
+    ProcessorDAO processorDAO;
+
+    @Inject
+    ConnectorsDAO connectorsDAO;
+
+    @Inject
+    ConnectorWorker connectorWorker;
```

Comment:
I'm not sure about the lifecycle of the workers. Sometimes they're injected while using the Vertx loop and sometimes they're injected directly?

Reply:
The entry-point is `handleWork(Work)`. Everything else is "internal".

`ConnectorWorker` is injected into `ProcessorWorker` because of our tightly coupled hard-coding.

That said; the work to provision or de-provision the dependencies for a `Processor` will always include the `Connector` and whilst we keep this recorded as a single item of work we will always need some form of chaining work... arguably we could also dispense with `ConnectorWorker` all together and just copy and paste the code into `ProcessorWorker`. It would do the same thing. However IMO keeping them separate makes in more simple: Class A does work for A and Class B does work for B... Also, if we ever move to provisioning the two things in parallel, we'd need separate Workers.. all that is missing is creating two Worker records instead of one (and a tiny change to `ConnectorWorker` to consume the VertX event for Connector work.

.. and there I go again.. waffling on :smiley:

Reply:
New commit added for "single entry point".